modules/payment/commerce_payment.module

```diff
@@ -125,12 +139,16 @@ function commerce_payment_create_field_definitions(array $definitions) {
  *
  * @param array $definitions
  *   The field definitions.
+ * @param bool $delete_storage
+ *   If set TRUE, field storage definition will be deleted also
  */
-function commerce_payment_delete_field_definitions(array $definitions) {
+function commerce_payment_delete_field_definitions(array $definitions, $delete_storage = FALSE) {
   $field_storage_definition_listener = \Drupal::service('field_storage_definition.listener');
   $field_definition_listener = \Drupal::service('field_definition.listener');
   foreach ($definitions as $definition) {
-    $field_storage_definition_listener->onFieldStorageDefinitionDelete($definition);
```

Comment:
Ignoring the big change, there is a subtle bug-fix in here as calling `onFieldStorageDefinitionDelete` before `onFieldDefinitionDelete` would result in database errors as Drupal tries to update tables that no longer exist.